doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -81,78 +65,75 @@ Flags, Options and Tables
 Coq has many settings to control its behavior.  Setting types include flags, options
 and tables:
 
-* A :production:`flag` has a boolean value, such as :flag:`Asymmetric Patterns`.
-* An :production:`option` generally has a numeric or string value, such as :opt:`Firstorder Depth`.
-* A :production:`table` contains a set of strings or qualids.
+* A *flag* has a boolean value, such as :flag:`Asymmetric Patterns`.
+* An *option* generally has a numeric or string value, such as :opt:`Firstorder Depth`.
+* A *table* contains a set of strings or qualids.
 * In addition, some commands provide settings, such as :cmd:`Extraction Language`.
 
 .. FIXME Convert "Extraction Language" to an option.
 
 Flags, options and tables are identified by a series of identifiers, each with an initial
 capital letter.
 
-.. cmd::  Set @flag
+.. cmd:: Set @setting_name {? {| @int | @string } }
    :name: Set
 
-   Sets :token:`flag` on.
+   .. insertprodn setting_name setting_name
 
-.. cmd:: Unset @flag
-   :name: Unset
-
-   Sets :token:`flag` off.
+   .. prodn::
+      setting_name ::= {+ @ident }
 
-.. cmd:: Test @flag
+   If :n:`@setting_name` is a flag, no value may be provided; the flag
+   is set to on.
+   If :n:`@setting_name` is an option, a value of the appropriate type
+   must be provided; the option is set to the specified value.
 
-   Prints the current value of :token:`flag`.
+   This command supports the :attr:`local`, :attr:`global` and :attr:`export` attributes.
+   They are described :ref:`here <set_unset_scope_qualifiers>`.
 
+   .. exn:: There is no option @setting_name.
 
-.. cmd:: Set @option {| @num | @string }
-   :name: Set @option
+      This message also appears for unknown flags.
 
-   Sets :token:`option` to the specified value.
+.. cmd:: Unset @setting_name
+   :name: Unset
 
-.. cmd:: Unset @option
-   :name: Unset @option
+   If :n:`@setting_name` is a flag, it is set to off.  If :n:`@setting_name` is an option, it is
+   set to its default value.
 
-   Sets :token:`option` to its default value.
+   This command supports the :attr:`local`, :attr:`global` and :attr:`export` attributes.
+   They are described :ref:`here <set_unset_scope_qualifiers>`.
 
-.. cmd:: Test @option
+.. cmd:: Test @setting_name {? for {+ {| @qualid | @string } } }
 
-   Prints the current value of :token:`option`.
+   If :n:`@setting_name` is a flag or option, prints its the current value.
+   If :n:`@setting_name` is a table: if a value is specified, reports whether
```

Comment:
One or several values are provided. And the description should be clearer on the fact that this is done with the `for` clause.